killerbee/dev_template.py

```diff
@@ -0,0 +1,169 @@
+from typing import Optional
+from typing import Dict
+from typing import Union
+from typing import Any
+
+import os
+import time
+import struct
+import time 
+from datetime import datetime, timedelta 
+from .kbutils import KBCapabilities, makeFCS 
+from .GoodFETCCSPI import GoodFETCCSPI
+
+class APIMOTE:
+    def __init__(self, dev: str) -> None:
+
+        self._channel: Optional[int] = None
+        self._page: int = 0
+        self.handle: Optional[Any] = None
+        self.dev: str = dev
+
+        self.handle = GoodFETCCSPI()
+        self.handle.serInit(port=self.dev)
+        self.handle.setup()
+
+        self.__stream_open: bool = False
+
+        self.capabilities: KBCapabilities = KBCapabilities()
+        self.__set_capabilities()
+
+    def __set_capabilities(self) -> None:
+        self.capabilities.setcapab(KBCapabilities.NONE, False)
+
+        self.capabilities.setcapab(KBCapabilities.FREQ_2400, True)
+        self.capabilities.setcapab(KBCapabilities.FREQ_900, False)
+        self.capabilities.setcapab(KBCapabilities.FREQ_863, False)
+        self.capabilities.setcapab(KBCapabilities.FREQ_868, False)
+        self.capabilities.setcapab(KBCapabilities.FREQ_870, False)
+        self.capabilities.setcapab(KBCapabilities.FREQ_915, False)
+
+        self.capabilities.setcapab(KBCapabilities.SNIFF, True)
+        self.capabilities.setcapab(KBCapabilities.SETCHAN, True)
+        self.capabilities.setcapab(KBCapabilities.INJECT, True)
+        self.capabilities.setcapab(KBCapabilities.SELFACK, False)
+        self.capabilities.setcapab(KBCapabilities.PHYJAM, False)
+        self.capabilities.setcapab(KBCapabilities.PHYJAM_REFLEX, False)
+        self.capabilities.setcapab(KBCapabilities.SET_SYNC, False)
+        return
+
+    def close(self) -> None:
+        
+        if self.handle is None:
+            raise Exception("Handle does not exist");
+
+        self.handle = None
+
+    def check_capability(self, capab: int) -> bool:
+        return self.capabilities.check(capab)
+
+    def get_capabilities(self) -> Dict[int, bool]:
+        return self.capabilities.getlist()
+
+    def get_dev_info(self) -> list[Union[str, Any]]:
+        return [self.dev, "GoodFET Apimote", ""]
```

Comment:
Probably change Apimote stuff to be generic name or TODO all over this file.

Reply:
noted. thanks